### ISSUE TITLE
Smooth samples

### ISSUE THREAD

**Issue by philipphennig** [AlexImmer/Laplace#91]
Hi all (not sure if you’re all still hanging out on slack),

I’ve just had a bit of fun playing with `laplace-torch` again to make a quick plot. One thing that keeps being a bit of a nuisance is that predictive plots with `link_approx="mc"` look “frayed”. I realised just now that this is because you're using different samples for each input location. Of course, the plot would be smooth if the samples were all the same for each input. 

This seems to be down to the code around [line 540 in baselaplace.py](https://github.com/AlexImmer/Laplace/blob/9a37ed7445d485e4c4c13900d8dd509976d7b55e/laplace/baselaplace.py#L539)

which calls a `torch.dist.normal.sample` . Wouldn’t it be better to draw just `n_samples` standard Gaussian RVs here and then broadcast those same samples to all locations in the input, multiplying with the sqrt-covariance and adding the local mean? This would have three advantages:

* it should be faster (no need to draw an insane number of RVs)
* the predictive then inherits the smoothness of the underlying logit layer. Can thus be used as an objective for all sorts of fun
* for the same reason, plots look nicer :) 

Come to think of it, it would be even nicer if those samples could be actively triggered, so people can keep them constant if they want to optimize (for active learning)

What do you think? 

Best,

Philipp

**Comment by AlexImmer** [comment]
Hi Philipp, thanks for the great suggestion. I agree with all the points you made and implemented the corresponding changes. The plots definitely look much nicer and it should be significantly faster. In case you want to control the seed of the samples, you will be able to pass a `torch.Generator` to the sampling and predictive function.

**Comment by philipphennig** [comment]
Awesome, thanks! I'll definitely try it out once this is merged into `main`. Thanks for taking care of this!
